Sources/PointFree/Account/Invite.swift

```diff
@@ -0,0 +1,158 @@
+import Css
+import Either
+import Foundation
+import Html
+import HtmlCssSupport
+import HttpPipeline
+import HttpPipelineHtmlSupport
+import Optics
+import Prelude
+import Styleguide
+import Tuple
+
+let showInviteMiddleware =
+  requireTeamInvite
+    <| currentUserMiddleware
+    >-> writeStatus(.ok)
+    >-> respond(showInviteView.contramap(lower))
+
+let revokeInviteMiddleware =
+  requireTeamInvite
+    <<< requireUser
+    <| { conn in
+      AppEnvironment.current.database.deleteTeamInvite(get2(conn.data).id)
+        .run
+        .flatMap(const(conn |> redirect(to: path(to: .team))))
```

Comment:
sometimes these middlewares can compose nicely. here we stack two transformers for requiring the team invite (from the `id` passed from the route), then requiring the user, and then we can just inline a middleware for doing the delete logic.